### DISCUSSION TITLE
Radsec Limitations and Benefits of RadsecProxy in Production

### DISCUSSION THREAD

**Discussion by emileswarts** [radsecproxy/radsecproxy#87] [Q&A] [answered]
Hi,

I'm implementing a 802.1x authentication service with FreeRadius which has RADSEC enabled where supported.
I've successfully tested Radsec on various networking equipment including Juniper, Meraki, and TPLink.

For automated testing I use eapol_test and have RadsecProxy running locally. RadsecProxy then points to the running service to convert UDP to TCP, to mimic production traffic. We are expecting this service to scale to support 100k+ devices spread across the UK.

I came across a note about the limitations of Radsec [here](https://www.juniper.net/documentation/us/en/software/junos/user-access/topics/task/radsec-configuring.html#:~:text=Due%20to%20limitations%20of%20the,255%20RADIUS%20messages%20in%20flight.) and I'd like to understand:

1. Will implementing RadsecProxy in production have any effect on this limit?
2. Is hitting this limit common? Have not found too many people online that ran into it. I was able to do 70 authentications per second using eapol_test + RadsecProxy + FreeRadius when performance testing.
3. Is there any benefit to deploying RadsecProxy over going straight from the authenticator to the server directly if we do not need to support existing authentication servers?

Emile

**Comment by fmauchle** [comment, marked as answer]
1. The limit described in the linked document is a fundamental limit of the radius protocol. It is a per connection limit, independent of the underlying transport (tcp/tls or udp - when considering a src/dst port tuple as a 'connection' for udp). There is however a limitation in radsecproxy that it currently can't open additional connections if needed. This is planned to be added in the future.
2. Under normal operating conditions, it is rather unlikely - as long as the authenticating radius server is quick to respond. But if the radius server is slow, or has a long RTT (over all radius proxy hops, not just IP hops), then you might hit this limit.
3. This very much depends on what features your authenticator provides and which you need. Radsecproxy is designed to add radsec capabilities to authenticators or servers that don't implement it, and to do realm routing for radius federations. If you don't need any of this, I guess there won't be much benefit.

**Reply by emileswarts** [reply]
Excellent, makes sense. 

Thanks
